adapters/x509_openssl.c

```diff
@@ -13,7 +13,7 @@
 
 #ifdef __APPLE__
     #ifndef EVP_PKEY_id
-        #define EVP_PKEY_id(evp_key) evp_key->type
+	#define EVP_PKEY_ID(evp_key) EVP_PKEY_get1_EC_KEY(evp_key)
```

Comment:
I ran into this issue myself - at this point we are only testing with OpenSSL 1.0 for OSX.

Would this work for RSA keys as well?